train.py

```diff
@@ -27,14 +28,33 @@ def makedir_if_not_exists(dir: str):
         os.makedirs(dir)
 
 
-def preprocess_dataset(dataset_name: str, batch_size: int, num_workers: int, *, overfit_single_batch: bool = False):
-    train_ds = load_dataset(dataset_name, split="train")
-    train_ds = MidiDataset(train_ds)
+def preprocess_dataset(dataset_name: str | list[str], batch_size: int, num_workers: int, *, overfit_single_batch: bool = False):
+    hf_token = os.environ["HUGGINGFACE_TOKEN"]
 
-    val_ds = load_dataset(dataset_name, split="validation")
-    val_ds = MidiDataset(val_ds)
+    if isinstance(dataset_name, (list, ListConfig)):
```

Comment:
I think we shouldn't support different types for a config value. In this case I'd prefer to always have a list, and if we want to use a single dataset, it can be a one-element list.